includes/admin/settings/class-wc-admin-settings-woorule.php

```diff
@@ -350,6 +333,33 @@ public static function order_status_changed($id, $status = '', $new_status = '')
                         );
                     }
 
+                    $custom_fields = [];
+
+                    if(get_option($rule['custom_fields']['id'])) {
+                        $cf = json_decode(get_option($rule['custom_fields']['id']));
+                        
+                        foreach($cf as $field) {
+                            
+                            if($field->attribute){
+
+                                if($field->source == 'user') {
+                                    $v = get_user_meta($order->user_id, $field->attribute, true);
+                                    $k = 'Subscriber.'.$field->attribute;
+                                } else {
+                                    $v = get_post_meta($order->id, $field->attribute, true);
+                                    $k = 'Order.'.$field->attribute;
+                                }
+
+                                if($v) {
+                                    array_push($subscription['subscribers']['fields'], array('key' => $k, 'value' => $v) );
+                                }
+                            }
+                        }
+                    }
+
+                    if($custom_fields) {
```

Comment:
Please, help me understand this part:) 
1) On line 336 we declare $custom_fields as an empty array and then we don't modify it. But on the line 360, we put $custom_fields inside if statement which always results in false. Is it expected?
2) Is it possible that on this step we will push fields that were already declared inside the `$subscription['subscribers']['fields']` as default (ex: Subscriber.FirstName)? Or it will be Subscriber.FirstName and Subscriber.first_name? If field names are identical we will get an error on Rule side as far as duplicated fields in API requests are not allowed. But if we push different field names, does it make sense to store identical data in different fields (FirstName and first_name)? 

Correct me if I'm wrong :slightly_smiling_face:

Reply:
Yes, you are correct. That's the part of the dev stuff i have not cleaned up.